python_modules/libraries/dagster-aws/dagster_aws/redshift/resources.py

```diff
@@ -272,67 +275,69 @@ def execute_queries(
-        "port": Field(
-            IntSource, description="Redshift port", is_required=False, default_value=5439
-        ),
-        "user": Field(
-            StringSource,
-            description="Username for Redshift connection",
-            is_required=False,
-        ),
-        "password": Field(
-            StringSource,
-            description="Password for Redshift connection",
-            is_required=False,
-        ),
-        "database": Field(
-            StringSource,
-            description=(
-                "Name of the default database to use. After login, you can use USE DATABASE"
-                " to change the database."
-            ),
-            is_required=False,
-        ),
-        "autocommit": Field(
-            bool,
-            description=(
-                "None by default, which honors the Redshift parameter AUTOCOMMIT. Set to "
-                "True or False to enable or disable autocommit mode in the session, respectively."
-            ),
-            is_required=False,
-        ),
-        "connect_timeout": Field(
-            int,
-            description="Connection timeout in seconds. 5 seconds by default",
-            is_required=False,
-            default_value=5,
+
+    host: str = PyField(..., description="Redshift host")
+    port: int = PyField(5439, description="Redshift port")
+    user: Optional[str] = PyField(None, description="Username for Redshift connection")
+    password: Optional[str] = PyField(None, description="Password for Redshift connection")
+    database: Optional[str] = PyField(
+        None,
+        description=(
+            "Name of the default database to use. After login, you can use USE DATABASE to change"
+            " the database."
         ),
-        "sslmode": Field(
-            str,
-            description=(
-                "SSL mode to use. See the Redshift documentation for more information on usage:"
-                " https://docs.aws.amazon.com/redshift/latest/mgmt/connecting-ssl-support.html"
-            ),
-            is_required=False,
-            default_value="require",
+    )
+    autocommit: Optional[bool] = PyField(None, description="Whether to autocommit queries")
+    connect_timeout: int = PyField(
+        5, description="Timeout for connection to Redshift cluster. Defaults to 5 seconds."
+    )
+    sslmode: str = PyField(
+        "require",
+        description=(
+            "SSL mode to use. See the Redshift documentation for reference:"
+            " https://docs.aws.amazon.com/redshift/latest/mgmt/connecting-ssl-support.html"
         ),
-    }
+    )
+
+    def create_resource(self, context: InitResourceContext) -> RedshiftClient:
+        return RedshiftClient(context)
+
+
+class FakeRedshiftResource(RedshiftResource):
+    def create_resource(self, context: InitResourceContext) -> FakeRedshiftClient:
+        return FakeRedshiftClient(context)
```

Comment:
IMO this should not inherit from `RedshiftResource` having config on the fake doesn't make too much sense